nix/magic-folder.nix

```diff
@@ -17,7 +17,20 @@ buildPythonPackage rec {
   ];
 
   postPatch = ''
-    PATH="$PATH:${git}/bin" ${python}/bin/python setup.py update_version
+    # Generate _version.py ourselves since we can't rely on the Python code
+    # extracting the information from the .git directory that is likely not
+    # available.
+    cat > src/magic_folder/_version.py <<EOF
+
+# This _version.py is generated from metadata by nix/magic-folder.nix.
+
+__pkgname__ = "magic-folder"
+real_version = "${version}"
+full_version = "${version}"
+branch = "main"
+verstr = "${version}"
```

Comment:
Looking at this, it appears that none of these are used (I think we inherited them from tahoe). I'd be inclined to remove all of them, and just set `__version__` directly (here and/or in `setup.py`).

Reply:
That seems fine to me and maybe also related to meejah's comment on the linked issue.  It seems beyond the scope of this effort though.  I just mirrored what setup.py is doing here.

Reply:
True, though if we go with the option I suggested above of allowing `_version.py` to not exists, that would address both issues, without being out-of-scope.

Reply:
I'd just as soon skip the extra rounds of review while we converge on a satisfactory replacement for `_version.py` though :)

Reply:
:shrug: The important thing to me here is to not embed a weird (i.e. incompatible with "upstream") version in the python package.

Reply:
The version numbers now match.